pciutils: use database from hwinfo which is easier to update

iblech opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Closes #112644

I foresee one problem with the pull request in its current form: pciutils has platforms set to unix, but hwdata, the new additional native build input, has platforms set to linux.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

iblech committed:
pciutils: use database from hwinfo which is easier to update

vcunat commented on pkgs/tools/system/pciutils/default.nix:
```diff
@@ -11,10 +12,18 @@ stdenv.mkDerivation rec {
+    # See #112644 for discussion of alternative solutions.
+    cp --reflink=auto ${hwdata}/share/hwdata/pci.ids pci.ids
+  '';
+
```
If it's done here, I don't think reflink really helps.  For it to make sense, I can imagine instead replacing `$out/share/pci.ids` during `postInstall` or something similar.

iblech replied:
Oh, you're right. I'll fix that in the imminent force-push!

vcunat commented:
I'm not sure how useful pciutils is on other platforms (e.g. darwin), but hwdata is just fixed-output data, so I can't see why it should be using such a restricted `meta.platforms`.

iblech commented:
Right. Should I submit a pull request to loosen hwdata's platform restriction to `platforms.all`?

vcunat commented on pkgs/tools/system/pciutils/default.nix:
```diff
@@ -11,7 +12,7 @@ stdenv.mkDerivation rec {
   };
 
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config hwdata ];
```
The dependency here is useless.  (and technically it shouldn't be a native but a normal dependency)

iblech replied:
Ah right. I agree it's unnecessary in nativeBuildInputs, as the compilation process doesn't require hwdata. However, it shouldn't be added as a normal dependency, right? Else we draw in the full closure of hwdata which we don't require. We need just one file from hwdata.

vcunat replied:
It wouldn't add a runtime dependency; those are computed by scanning the results.  But we don't need to add it anywhere.  It will become a build-time dependency just because it's used in some of the phases.

iblech replied:
Oh, I see! Thank you for pointing that out, didn't know that. Fixed in the new force-push.

SuperSandro2000 commented:
> Right, and thank you for the quick review! Should I submit a pull request to loosen hwdata's platform restriction to `platforms.all`?

We can do that here. No need for another PR.

SuperSandro2000 commented on pkgs/tools/system/pciutils/default.nix:
```diff
@@ -30,8 +31,13 @@ stdenv.mkDerivation rec {
-  # Get rid of update-pciids as it won't work.
-  postInstall = "rm $out/sbin/update-pciids $out/man/man8/update-pciids.8";
+  postInstall = ''
+    # Get rid of update-pciids as it won't work ...
```
```suggestion
    # Remove update-pciids as it won't work on nixos
```

SuperSandro2000 commented on pkgs/tools/system/pciutils/default.nix:
```diff
@@ -30,8 +31,13 @@ stdenv.mkDerivation rec {
+    # Get rid of update-pciids as it won't work ...
+    rm $out/sbin/update-pciids $out/man/man8/update-pciids.8
+
+    # ... but also update database with newer version from hwdata
```
```suggestion
    # use database from hwdata instead
```

SuperSandro2000 commented on pkgs/tools/system/pciutils/default.nix:
```diff
@@ -30,8 +31,13 @@ stdenv.mkDerivation rec {
+    rm $out/sbin/update-pciids $out/man/man8/update-pciids.8
+
+    # ... but also update database with newer version from hwdata
+    cp --reflink=auto ${hwdata}/share/hwdata/pci.ids $out/share/pci.ids
```
```suggestion
    ln -s ${hwdata}/share/hwdata/pci.ids $out/share/pci.ids
```

iblech replied:
But then pciutils will depend on the full closure of hwdata, right? We wanted to avoid that over at #112644. I'll add a comment to the source!

iblech committed:
hwdata: remove restriction to platforms.linux

hwdata is just a collection of (plain-text) databases and hence makes
sense on any platform. The impetus for this commit was #112644, where we
wanted to change pciutils so that it uses the database from hwdata. At
this time, pciutils was marked as platforms.unix but hwdata only as
platforms.linux.

vcunat commented:
Well, except it's too big to merge directly to master.  I didn't retarget it myself because of containing also a few commits from `staging..master` which would make the GitHub UI confusing.

iblech commented:
I'll try to rebase and retarget, let's see whether I manage to get that done :-)

vcunat commented:
Looks OK.  It's red because of [an unrelated evaluation error](https://gist.github.com/GrahamcOfBorg/63582ce536a886ab6cbbb6d6deecc3dd).

cole-h commented:
The eval error might be fixed by rebasing on the latest `staging` (e.g. `git pull --rebase origin staging`) -- the double definition of `DRM_AMD_DC_DCN3_0` has been fixed.
